src/vs/workbench/services/dialogs/browser/remoteFileDialog.ts

```diff
@@ -179,6 +179,7 @@ export class RemoteFileDialog {
 			this.filePickBox = this.quickInputService.createQuickPick<FileQuickPickItem>();
 			this.filePickBox.matchOnLabel = false;
 			this.filePickBox.autoFocusOnList = false;
+			this.filePickBox.ignoreFocusOut = true;
```

Comment:
There is a user setting for this. Doesn't that cover the ask?

Reply:
The ask was to make it not just dismiss when you click away from it, similar to the system dialog, and to always require a deliberate action to dismiss it. Since we cannot get that behavior by default with the user setting, we need to add this.

Reply:
I would only replicate the system dialog's behavior where that aligns with VS Code, not the other way around. 🤔